packages/edit-site/src/components/dataviews/view-list.js

```diff
@@ -427,12 +427,13 @@ function ViewList( {
 			} );
 		},
 		onGlobalFilterChange: ( value ) => {
-			onChangeView( { ...view, search: value, page: 0 } );
+			onChangeView( { ...view, search: value, page: 1 } );
```

Comment:
Apparently, this value is not being used – or is ignored once the own `Search` component updates it to 1 [here](https://github.com/WordPress/gutenberg/blob/trunk/packages/edit-site/src/components/dataviews/search.js#L25). I thought I'd change it anyway, though this needs a separate further investigation.